### PR TITLE
Issue #ED-3071 feat : 7.0.0 framework skip onboarding popup

### PR DESCRIPTION
**Pull request by mithun30052001**
# SunbirdEd - Portal
---

### Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
---

#### Example
- [x] Applicable
- [ ] Not applicable

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/client/src/app/modules/core/services/user/user.service.ts [hunk 1]
```diff
@@ -485,6 +487,14 @@ export class UserService {
       return response;
     }));
   }
+
```
Comment (princegupta1131) [inline src/app/client/src/app/modules/core/services/user/user.service.ts:490]: @mithun30052001  @description - add in common place before the method  and fix the indentation

File: src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts [hunk 2]
```diff
@@ -56,10 +58,18 @@ export class LocationSelectionComponent implements OnInit, OnDestroy, AfterViewI
 
   ngOnInit() {
     this.popupControlService.changePopupStatus(false);
+    /**
```
Comment (princegupta1131) [inline src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts:61]: @mithun30052001 add the @description - above the method for better readability

File: src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.ts [hunk 3]
```diff
@@ -48,10 +50,22 @@ export class OnboardingUserSelectionComponent implements OnInit, OnDestroy {
...
-    private toasterService: ToasterService
+    private toasterService: ToasterService,
+    private popupControlService: PopupControlService
   ) { }
 
   ngOnInit() {
+    /**
+      * @description - Here we enable/disable the usertype popup based on the isvisible value returned from the form config request 
+      *                along with setting up the required localstorage values in the case usertype popup is disabled
+    */
```
Comment (princegupta1131) [inline src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.ts:61]: @mithun30052001  refactor this code  if(this.isUserTypeEnabled === false){
if(!(this.isUserTypeEnabled)) try this

File: src/app/client/src/app/app.component.ts [hunk 4]
```diff
@@ -313,11 +317,38 @@ export class AppComponent implements OnInit, OnDestroy {
...
+      component: "portal"
+    };
+    this.formService.getFormConfig(formReadInputParams).subscribe(
+      (formResponsedata) => {
+        if (formResponsedata) {
+          this.popupControlService.setOnboardingData(formResponsedata);
+          this.isOnboardingEnabled= formResponsedata.onboardingPopups? formResponsedata.onboardingPopups.isVisible : true;
+          this.isFWSelectionEnabled = formResponsedata.frameworkPopup? formResponsedata.frameworkPopup.isVisible : true;
+          this.isUserTypeEnabled =formResponsedata.userTypePopup? formResponsedata.userTypePopup.isVisible: true;
+          if(this.isOnboardingEnabled === false || this.isFWSelectionEnabled === false){
```
Comment (princegupta1131) [inline src/app/client/src/app/app.component.ts:341]: @mithun30052001  refactor this line of code if(this.isOnboardingEnabled === false || this.isFWSelectionEnabled === false){

File: src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts [hunk 5]
```diff
@@ -56,10 +58,18 @@ export class LocationSelectionComponent implements OnInit, OnDestroy, AfterViewI
...
+      * @description - Here we enable/disable the location popup based on the isvisible value returned from the form config request
+    */
+    this.popupControlService.getOnboardingData().subscribe((formResponsedata)=>{
+      this.isLocationEnabled= formResponsedata.locationPopup? formResponsedata.locationPopup.isVisible : true;
+    })
     this.sbFormLocationSelectionDelegate.init(this.deviceProfile, this.showModal, this.isStepper )
       .catch(() => {
         this.closeModal();
-        this.toasterService.error(this.resourceService.messages.fmsg.m0049);
+        if(this.isLocationEnabled === true){
```
Comment (princegupta1131) [inline src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts:70]: @mithun30052001 refactor this if(this.isLocationEnabled === true){

File: src/app/client/src/app/service/popup-control.service.ts [hunk 6]
```diff
@@ -8,11 +8,18 @@ export class PopupControlService {
...
   public checkPopupStatus = this.popupSource.asObservable();
-
+  private onboardPopupSubject = new BehaviorSubject<any>(null);
   constructor() { }
 
   changePopupStatus(value: boolean) {
     this.popupSource.next(value);
   }
 
+  setOnboardingData(formData: any) {
```
Comment (princegupta1131) [inline src/app/client/src/app/service/popup-control.service.ts:18]: @mithun30052001  add the comment description

File: src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts [hunk 7]
```diff
@@ -56,10 +58,18 @@ export class LocationSelectionComponent implements OnInit, OnDestroy, AfterViewI
 
   ngOnInit() {
     this.popupControlService.changePopupStatus(false);
+    /**
+      * @description - Here we enable/disable the location popup based on the isvisible value returned from the form config request
```
Comment (princegupta1131) [inline src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts:62]: @mithun30052001 refactor this comment Here we enable/disable the location popup based on the isvisible value returned from the form config request

File: src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.ts [hunk 8]
```diff
@@ -48,10 +50,22 @@ export class OnboardingUserSelectionComponent implements OnInit, OnDestroy {
     private formService: FormService,
     private profileService: ProfileService,
     private userService: UserService,
-    private toasterService: ToasterService
+    private toasterService: ToasterService,
+    private popupControlService: PopupControlService
   ) { }
 
   ngOnInit() {
+    /**
```
Comment (princegupta1131) [inline src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.ts:58]: @mithun30052001 refactor this comment Here we enable/disable the usertype popup based on the isvisible value returned from the form config request 
      *                along with setting up the required localstorage values in the case usertype popup is disabled
    */

**Review by princegupta1131 — COMMENTED** [review]
refactor the comment description

File: src/app/client/src/app/app.component.ts [hunk 9]
```diff
@@ -313,11 +317,38 @@ export class AppComponent implements OnInit, OnDestroy {
     //       this.isStepperEnabled = true;
     //     }
     //     else { this.isPopupEnabled = true; }
-    //   }, error => { this.isPopupEnabled = true; });
+    //   }, error => { this.isPopupEnabled = true; });;
     this.isPopupEnabled = true;
   }
+
+  getOnboardingSkipStatus(){
```
[the remark this answers is not in the document]
Comment (mithun30052001) [reply src/app/client/src/app/app.component.ts]: Refactor done @princegupta1131

File: src/app/client/src/app/app.component.ts (hunk 4 above)
Comment (mithun30052001) [reply src/app/client/src/app/app.component.ts]: Refactor done @princegupta1131

File: src/app/client/src/app/modules/core/services/user/user.service.ts (hunk 1 above)
Comment (mithun30052001) [reply src/app/client/src/app/modules/core/services/user/user.service.ts:490]: Added in one common place @princegupta1131

File: src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.ts (hunk 8 above)
Comment (mithun30052001) [reply src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.ts:58]: comment refactored @princegupta1131

File: src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.ts (hunk 3 above)
Comment (mithun30052001) [reply src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.ts:60]: Refactor done @princegupta1131

File: src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts (hunk 5 above)
Comment (mithun30052001) [reply src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts]: Refactor done @princegupta1131

File: src/app/client/src/app/service/popup-control.service.ts (hunk 6 above)
Comment (mithun30052001) [reply src/app/client/src/app/service/popup-control.service.ts:21]: Comments added @princegupta1131

**Review by princegupta1131 — COMMENTED** [review]
check and refactor the comment description

**Review by rajeevsathish — APPROVED** [review]

**Pull request merged**